source/common/http/http2/codec_impl.cc

```diff
@@ -347,6 +348,11 @@ MetadataDecoder& ConnectionImpl::StreamImpl::getMetadataDecoder() {
 }
 
 void ConnectionImpl::StreamImpl::onMetadataDecoded(MetadataMapPtr&& metadata_map_ptr) {
+  if (decoder_destroyed_) {
+    // nghttp2 returns error, and ConnectionManager will call resetAllStreams(). Avoid referring to
+    // decoder_.
```

Comment:
If decoder_ is unsafe to refer to, should it be nulled out rather than being tracked as a separate boolean?  This seems fragile.

Reply:
Fixed. Thanks!